docs/source/miscellaneous/doc/adr/0016-use-tickets-encoding-for-transactions.md

```diff
@@ -682,19 +682,122 @@ C1s (and C2s) never run concurrently with C4s.
+reads would all go to a single replication group in transactions1. This will substantially reduce read loads on the
+replication group that was responsible for the data in `_transactions`, though note that this effect may take a while
+to observe, depending on one's read patterns. Data written under the transactions1 scheme will still need to be read
+from the specific replication group.
+
+The amount of data read is also quite a bit smaller - this is expected, as the keys and values used in the
+``_transactions2`` table are generally more compact than that in ``_transactions``.
+
+Our changes to the ``CellLoader`` may also improve performance for dynamic column tables (other than ``_transactions2``)
+where query patterns involve loading distinct columns across multiple rows; though the team is not currently aware
+of any specific examples.
 
 ### Operational Concerns
 
+Unfortunately, transactions2 introduces some complexity in terms of operation. Most obviously, reading and writing
+to the table outside of through an AtlasDB client becomes more difficult. There are also several other issues to be
+aware of.
+
 #### Backup and Restore
 
-#### Cassandra Dependencies
+AtlasDB backups are effectively done at a single logical point in time - the *backup timestamp*. This is taken before
+the underlying key-value service is backed up; we then take a *fast-forward timestamp* after this is complete so that
+we are able to distinguish data that was written during our backup, and new data written after our restore (that could
+happen if, for example, timestamp state was stored in the key-value service). As part of restoring, we also delete all
+entries in the transactions table that committed after our backup timestamp.
+
+The addition of the transactions2 table doesn't really cause issues; the bigger challenge here is working with the
+coordination service, which needs to be consistent with the usage of the transaction tables. These tables need special
+treatment - otherwise, there can be unexpected situations like the following:
+
+1. We get a backup timestamp at time 1000.
+2. At timestamp 1100, we switch to use transactions2.
+3. We backup the coordination table. At this point, the range map for transaction schema versions would be
+   ``{[1, 1100) = 1, [1100, +∞) = 2}``.
+4. At timestamp 1200, we switch back to transactions1.
+5. A write transaction T that writes to table X starts at timestamp 1230 and commits at timestamp 1250.
+6. We backup table X (note that not all KVS backup schemes are guaranteed to be point-in-time backups!)
+7. We get a fast-forward timestamp at time 1300.
+8. On restore, our copy of the coordination table suggests that we want to clean-transactions-range for values in
+   transactions1 from 1000 to 1100 (exclusive), and values in transactions2 from 1100 onwards. However, the values
+   written by T will still remain visible in this case, when they shouldn't be (because they were after the
```

Comment:
But it's not actually visible because coordination would tell us that for this timestamp the schema version is 2 (either by extending the bound on the existing entry, or changing the version, but starting with a timestamp that would be greater than the fast-forward timestamp), no?